compass_metrics/git_metrics.py

```diff
@@ -165,25 +264,27 @@ def git_pr_linked_ratio(client, git_index, pr_index, date, repos_list):
                 commit_pr_cout += 1
             else:
                 commit_message_dict[commit_hash] = 0
-    if commit_count>0:
-        # return len(commit_all_message), commit_pr_cout, commit_pr_cout/len(commit_all_message)
-        git_pr_linked_ratio = commit_pr_cout/len(commit_all_message)
-    else:
-        # return 0, None, None
-        git_pr_linked_ratio = None
-       
+
     result = {
-        'git_pr_linked_ratio': git_pr_linked_ratio
+        'commit_pr_linked_ratio': commit_pr_cout/len(commit_all_message) if commit_count > 0 else None,
+        'commit_pr': len(commit_all_message),
+        'commit_pr_linked': commit_pr_cout if commit_count > 0 else None
     }
     return result
 
 
-def LOC_frequency(client, git_index, date, repos_list, field='lines_changed'):
-    query_LOC_frequency = get_uuid_count_query(
-        'sum', repos_list, field, 'grimoire_creation_date', size=0, from_date=date-timedelta(days=90), to_date=date)
-    LOC_frequency = client.search(index=git_index, body=query_LOC_frequency)[
-        'aggregations']['count_of_uuid']['value']
+def lines_of_code_frequency(client, git_index, date, repos_list):
+    """ Determine the average number of lines touched (lines added plus lines removed) per week in the past 90 """
+    def LOC_frequency(client, git_index, date, repos_list, field='lines_changed'):
+        query_LOC_frequency = get_uuid_count_query(
+            'sum', repos_list, field, 'grimoire_creation_date', size=0, from_date=date-timedelta(days=90), to_date=date)
+        loc_frequency = client.search(index=git_index, body=query_LOC_frequency)[
+            'aggregations']['count_of_uuid']['value']
+        return loc_frequency/12.85
+
     result = {
-        'LOC_frequency': LOC_frequency/12.85
+        "lines_of_code_frequency": LOC_frequency(client, git_index, date, repos_list, 'lines_changed'),
+        "lines_add_of_code_frequency": LOC_frequency(client, git_index, date, repos_list, 'lines_added'),
+        "lines_remove_of_code_frequency": LOC_frequency(client, git_index, date, repos_list, 'lines_removed'),
     }
```

Comment:
![image](https://github.com/oss-compass/compass-metrics-model/assets/13914416/37ec2920-cdc7-465d-b87c-07bc3e25e54a)

Need to review the final stored fields' name try not to change, otherwise you need to explicitly state that this will involve data migration work!